### PR TITLE
libertinus: 6.9 -> 7.040

### PR DESCRIPTION
**Pull request by waldyrious**
Besides the version bump, the Nix expression was updated to conform to directory structure changes in the archive file, and to add extra documentation files that were not included before.

Also, the short and long descriptions were updated to match the current contents of the project's README.

Finally, the homepage and owner fields were changed to reflect the current location of the project.

/cc @dtzWill who contributed the previous updates (#31461, #46725, #61556 and #62240).

###### Motivation for this change

Update the Libertinus package to the latest version.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by waldyrious** [comment]
Not sure if this is something that can be addressed in this PR, but I noticed that the [search result page](https://search.nixos.org/packages?show=libertinus&query=libertinus) doesn't show the "Source" link to this expression file, unlike e.g. [this one](https://search.nixos.org/packages?show=kakoune&query=kakoune). If there's anything I can do to fix that, please let me know.

**Comment by waldyrious** [comment]
Also, for future reference: I got different results when trying to compute the sha256 of the package vs the format that the .nix file appears to expect. My first approach was to do

```console
$ sha256sum Libertinus-7.040.tar.xz 
7fe9f022722d1c1cc67dc2c28a110b3bb55bae3575196160d2422c89333b3850  Libertinus-7.040.tar.xz
```

...but since it was of a different length and format from the previous sha256 value (`0765a7w0askkhrjmjk638gcm9h6fcm1jpaza8iw9afr3sz1s0xlq`), I looked into the docs for hints. I didn't find anything about how to compute the sha256 value in the ["Quick Start to Adding a Package"](https://nixos.org/manual/nixpkgs/stable/#chap-quick-start) chapter (which is linked from the "no search results" page since https://github.com/NixOS/nixos-search/pull/256) but I searched for "sha256" and found an example in the ["How to install Cataclysm DDA"](https://nixos.org/manual/nixpkgs/stable/#how-to-install-cataclysm-dda) section (:shrug:), which I adapted as follows:

```console
$ nix-prefetch-url --unpack "https://github.com/alerque/libertinus/archive/v7.040.tar.gz"
unpacking...
[8.1 MiB DL]
path is '/nix/store/2k1x5jbv4xpsjb1dlqs387v6nqfsj345-v7.040.tar.gz'
0i68ka4s9lkf13km6gl2wgwscamhbq792arscp5wyha7l6qycq17
```

I'm leaving this for future reference, but also as feedback for project maintainers since this is my first contribution to this repo, and this was a stumbling block.

**Comment by Artturin** [comment]
nixpkgs-review fails
```

trying https://github.com/alerque/libertinus/archive/v7.040.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   126  100   126    0     0    400      0 --:--:-- --:--:-- --:--:--   400

100   731    0   731    0     0    933      0 --:--:-- --:--:-- --:--:--   933
100  944k    0  944k    0     0   549k      0 --:--:--  0:00:01 --:--:-- 1008k
100 3073k    0 3073k    0     0  1127k      0 --:--:--  0:00:02 --:--:-- 1579k
100 6875k    0 6875k    0     0  1851k      0 --:--:--  0:00:03 --:--:-- 2344k
100 8282k    0 8282k    0     0  2094k      0 --:--:--  0:00:03 --:--:-- 2611k
install: missing file operand
Try 'install --help' for more information.
```

**Comment by waldyrious** [comment]
@Artturin thanks for taking a look. Unfortunately I don't know how to fix that issue. Do you have any suggestions?

**Comment by Artturin** [comment]
It was downloading the source instead of the tar that contains the built files

apply this patch (with `git apply`) and it'll work, also we dont need the docs
```
diff --git a/pkgs/data/fonts/libertinus/default.nix b/pkgs/data/fonts/libertinus/default.nix
index a9417fa4abc..e36252890d9 100644
--- a/pkgs/data/fonts/libertinus/default.nix
+++ b/pkgs/data/fonts/libertinus/default.nix
@@ -1,22 +1,19 @@
-{ lib, fetchFromGitHub }:
+{ lib, fetchurl }:
 
 let
   version = "7.040";
-in fetchFromGitHub rec {
+in fetchurl rec {
   name = "libertinus-${version}";
+  url = "https://github.com/alerque/libertinus/releases/download/v${version}/Libertinus-${version}.tar.xz";
+  sha256 = "sha256-GT2kBfTCP1lf8SYG0z6iii/1482hx3uUNsPTCrM+DKQ=";
 
-  owner  = "alerque";
-  repo   = "libertinus";
-  rev    = "v${version}";
+  downloadToTemp = true;
+  recursiveHash = true;
 
   postFetch = ''
     tar xf $downloadedFile --strip=1
-    install -m444 -Dt $out/share/fonts/opentype static/OTF/*.otf
-    install -m444 -Dt $out/share/doc/${name}    *.txt
-    install -m444 -Dt $out/share/doc/${name}    *.md
-    install -m444 -Dt $out/share/doc/${name}    documentation/*
+    install -m644 -Dt $out/share/fonts/opentype static/OTF/*.otf
   '';
-  sha256 = "0i68ka4s9lkf13km6gl2wgwscamhbq792arscp5wyha7l6qycq17";
 
   meta = with lib; {
     description = "The Libertinus font family";
```

**Comment by Artturin** [comment]
> Also, for future reference: I got different results when trying to compute the sha256 of the package vs the format that the .nix file appears to expect. My first approach was to do
> 
> ```
> $ sha256sum Libertinus-7.040.tar.xz 
> 7fe9f022722d1c1cc67dc2c28a110b3bb55bae3575196160d2422c89333b3850  Libertinus-7.040.tar.xz
> ```
> 
> ...but since it was of a different length and format from the previous sha256 value (`0765a7w0askkhrjmjk638gcm9h6fcm1jpaza8iw9afr3sz1s0xlq`), I looked into the docs for hints. I found the following passage in in the ["Quick Start to Adding a Package"](https://nixos.org/manual/nixpkgs/stable/#chap-quick-start) chapter (which is linked from the "no search results" page since [NixOS/nixos-search#256](https://github.com/NixOS/nixos-search/pull/256)):
> 
> > You can use `nix-prefetch-url url` to get the SHA-256 hash of source distributions.
> 
> ...but it didn't explain how to actually do it, so I searched for "sha256" and found an example in the ["How to install Cataclysm DDA"](https://nixos.org/manual/nixpkgs/stable/#how-to-install-cataclysm-dda) section, which I adapted as follows:
> 
> ```
> $ nix-prefetch-url --unpack "https://github.com/alerque/libertinus/archive/v7.040.tar.gz"
> unpacking...
> [8.1 MiB DL]
> path is '/nix/store/2k1x5jbv4xpsjb1dlqs387v6nqfsj345-v7.040.tar.gz'
> 0i68ka4s9lkf13km6gl2wgwscamhbq792arscp5wyha7l6qycq17
> ```

usually i get the hash by just filling the sha256 field with 52 zeroes (`52i0<esc>` in vim) and then building the package `nix-build '.' -A libertinus` and copying the hash it gives you

**Comment by waldyrious** [comment]
Hi @Artturin, thanks for the hints and sorry for the delayed response. I must have mixed things up — for some reason I was under the impression that I had downloaded the source tarball and checked that it contained the built fonts, but of course now I see that wouldn't make sense. Apologies for the confusion.

I made the changes you proposed, and updated the commit message accordingly. Let me know if it looks good!

File: pkgs/data/fonts/libertinus/default.nix [hunk 1]
```diff
@@ -1,29 +1,29 @@
...
+{ lib, fetchurl }:
 
 let
-  version = "6.9";
-in fetchFromGitHub rec {
+  version = "7.040";
+in fetchurl rec {
   name = "libertinus-${version}";
+  url = "https://github.com/alerque/libertinus/releases/download/v${version}/Libertinus-${version}.tar.xz";
+  sha256 = "sha256-GT2kBfTCP1lf8SYG0z6iii/1482hx3uUNsPTCrM+DKQ=";
```
Comment (SuperSandro2000) [inline pkgs/data/fonts/libertinus/default.nix:8]:
```suggestion
  sha256 = "0z658r88p52dyrcslv0wlccw0sw7m5jz8nbqizv95nf7bfw96iyk";
```

Comment (waldyrious) [reply pkgs/data/fonts/libertinus/default.nix]: Thanks! :pray:

**Pull request merged**